xarray/coding/times.py

```diff
@@ -291,7 +279,12 @@ def cftime_to_nptime(times):
     times = np.asarray(times)
     new = np.empty(times.shape, dtype='M8[ns]')
     for i, t in np.ndenumerate(times):
-        dt = datetime(t.year, t.month, t.day, t.hour, t.minute, t.second)
+        try:
+            dt = pd.Timestamp(t.year, t.month, t.day, t.hour, t.minute,
```

Comment:
I switched to using `pd.Timestamp` here, because directly converting from `datetime.datetime` to `np.datetime64[ns]` did not raise an error for dates outside 1678 to 2262:
```
In [1]: from datetime import datetime; import numpy as np

In [2]: np.datetime64(datetime(1, 1, 1), 'ns')
Out[2]: numpy.datetime64('1754-08-30T22:43:41.128654848')
```